byteutils.nim

```diff
@@ -34,34 +34,38 @@ proc readHexChar*(c: char): byte {.noSideEffect, inline.}=
   else:
     raise newException(ValueError, $c & "is not a hexademical character")
 
-proc skip0xPrefix(hexStr: string): int {.noSideEffect, inline.} =
+template skip0xPrefix(hexStr: string): int =
```

Comment:
Why the switch to template here?

Reply:
Force inlining. I know the proc should be inlined anyway, but templates ensure it.

Reply:
I never had an issue with inline procs not being inlined. It's not implemented with a C inline pragma but with a NIm/C macro so they are always inlined.

It's easy to check in assembly or with a benchmark tool that traces function call.

Reply:
Er, this is not true. `.inline.` procs are compiled to definitions using `N_INLINE` from nimbase.h:
https://github.com/nim-lang/Nim/blob/devel/lib/nimbase.h#L95

Reply:
Ah thanks for the correction.